tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -625,7 +565,42 @@ def test_retrieve_object_tags(self, user_attr, expected_status, expected_count):
         assert response.status_code == expected_status
 
         if status.is_success(expected_status):
-            assert len(response.data) == expected_count
+            # Check the response, first converting from OrderedDict to regular dicts for simplicity.
+            assert response.data == {
+                "problem15": {
+                    "taxonomies": [
+                        {
+                            "name": "Life on Earth",
+                            "taxonomy_id": 1,
+                            "editable": True,
+                            "tags": [
+                                {
+                                    "name": "Life on Earth",
+                                    "value": "Mammalia",
+                                    "lineage": ["Eukaryota", "Animalia", "Chordata", "Mammalia"],
+                                },
+                                {
+                                    "name": "Life on Earth",
+                                    "value": "Fungi",
+                                    "lineage": ["Eukaryota", "Fungi"],
+                                },
+                            ]
+                        },
+                        {
+                            "name": "user AUthors",
+                            "taxonomy_id": 3,
+                            "editable": False,
+                            "tags": [
+                                {
+                                    "name": "User Authors",
+                                    "value": "test_user_1",
+                                    "lineage": ["test_user_1"],
+                                },
+                            ],
+                        }
+                    ],
+                },
+            }
```

Comment:
@yusuf-musleh If I make the "get object tags" API return data like this, it should make the unit tags overlay much simpler, right?

```python
            assert response.data == {
                "problem15": {
                    "taxonomies": [
                        {
                            "name": "Life on Earth",
                            "taxonomy_id": 1,
                            "editable": True,
                            "tags": [
                                {
                                    "name": "Life on Earth",
                                    "value": "Mammalia",
                                    "lineage": ["Eukaryota", "Animalia", "Chordata", "Mammalia"],
                                },
                                {
                                    "name": "Life on Earth",
                                    "value": "Fungi",
                                    "lineage": ["Eukaryota", "Fungi"],
                                },
                            ]
                        },
                        {
                            "name": "user AUthors",
                            "taxonomy_id": 3,
                            "editable": False,
                            "tags": [
                                {
                                    "name": "User Authors",
                                    "value": "test_user_1",
                                    "lineage": ["test_user_1"],
                                },
                            ],
                        }
                    ],
                },
            }
```

Reply:
@bradenmacdonald Absolutely! This would make things much simpler. In the currently implementation there's a bunch of stitching stuff together, having this would reduce the API calls needed, especially if the taxonomies returned included taxonomies that are "available" and not only limited to ones that have their tags applied to the object (unit).